Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -477,6 +477,51 @@ void TextureCacheBase::BindTextures()
   }
 }
 
+class ArbitraryMipmapDetector
+{
+public:
+  void AddLevel(u32 width, u32 height, u32 row_length, const u8* buffer)
+  {
+    ++levels;
+
+    std::array<u8, 4> avg = { 0 };
+    for (u32 i = 0; i < height; ++i)
+    {
+      const auto* row = buffer;
+      for (u32 j = 0; j < width; ++j)
+      {
+        avg[0] += row[0];
+        avg[1] += row[1];
+        avg[2] += row[2];
+        avg[3] += row[3];
+
+        row += 4;
+      }
+      
+      buffer += row_length;
+    }
+
+    for (size_t c = 0; c < avg.size(); ++c)
```

Comment:
My original comment put the `std::` on the `size_t` because there's a [proposal](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/p0657r0.pdf) to deprecate implicitly dumping things from `<cxxxxx>` headers (I didn't just toss it in there for fun :p).